### PR TITLE
[com_actionlogs][Security][RELEASE BLOCKER] Check for token before allowing purge

### PR DESCRIPTION
**Pull request by PhilETaylor**
@joomla/security 

Add missing token check for CSRF to purge action

Good, and important catch!

### CODE REVIEW

File: administrator/components/com_actionlogs/controllers/actionlogs.php [hunk 1]
```diff
@@ -116,6 +116,9 @@ public function exportLogs()
 	 */
 	public function purge()
 	{
+		// Check for request forgeries.
+		JSession::checkToken() or die(JText::_('JINVALID_TOKEN'));
```
Comment (SharkyKZ) [inline administrator/components/com_actionlogs/controllers/actionlogs.php]: Use `$this->checkToken()` please. See #12801.

Comment (PhilETaylor) [reply administrator/components/com_actionlogs/controllers/actionlogs.php]: its literally copy and pasted from further up the file!

**Review by wilsonge — APPROVED** [review]

File: administrator/components/com_actionlogs/controllers/actionlogs.php [hunk 2]
```diff
@@ -116,6 +116,9 @@ public function exportLogs()
 	 */
 	public function purge()
 	{
+		// Check for request forgeries.
+		$this->checkToken();
+
```
Comment (SharkyKZ) [inline administrator/components/com_actionlogs/controllers/actionlogs.php]: Remove tabs.

**Comment by PhilETaylor** [comment]
I think you are seeing things! This was a direct copy and paste in github in the browser.. I see no tabs?!?!

——
Sent from my iPhone - sorry - if needed I’ll send longer email from my desk later.
——

**Comment by SharkyKZ** [comment]
tabs

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on 81efaf74cbac508207a81a044b3f833c72c57322<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22705">issues.joomla.org/tracker/joomla-cms/22705</a>.</sub>

**Comment by zero-24** [comment]
Merging thanks.

**Pull request merged**